src/parser.js

```diff
@@ -375,7 +378,7 @@ function stateMachineStatementParser (statement, steps, { isStrict, dialect = 'g
       if (token.type === 'semicolon') {
         // SQLite and MSSQL require semi-colons inside the trigger. They signify the end of the trigger creation
         // with `END;`. This allows detection of that.
-        if (dialectsWithEnds.includes(dialect) && (statement.type === 'CREATE_TRIGGER' && !statement.canEnd)) {
+        if (dialectsWithEnds.includes(dialect) && (statementsWithEnds.includes(statement.type) && !statement.canEnd)) {
```

Comment:
Probably not worth doing now, but we should probably abstract these checks into a `function requiresEnd` or something like that

Reply:
agreed, I plan to do a handful of various clean-up stuff before 1.3.0 release that I'll fold this into.